### DISCUSSION TITLE
Impose a dirichlet boundary condition on one component of a vertor variable and ignore the other components

### DISCUSSION THREAD

**Discussion by bylore** [idaholab/moose#23994] [Q&A General] [answered]
hello，I want impose a dirichlet boundary condition on only one component of a vertor variable and ignore the other components. Is there any built-in boundary conditions that can be used in moose? 

It seems that [DirichletBC](https://mooseframework.inl.gov/source/bcs/DirichletBC.html) can impose dirichlet doundary condition on a scalar variable, and [VectorDirichletBC](https://mooseframework.inl.gov/source/bcs/VectorDirichletBC.html) can deal with vector variables but cannot pick a specific component.

Thanks for your help！

**Comment by doquang** [comment, marked as answer]
I think, the AD version of VectorNodalBC as `ADVectorDirichletBC` and `ADVectorFunctionDirichletBC` has params: `set_x_comp`, `set_y_comp`, `set_z_comp`. To ignore one or two components of the vector, just set those parameter to false.

**Reply by bylore** [reply]
Thanks for your reply. I found these parameters in ADVectorMatchedValueBC.

**Reply by bylore** [reply]
Yes，ADVectorFunctionDirichletBC works well for my issue, thank you
